server/tserver/src/main/java/org/apache/accumulo/tserver/log/CreateEmptyWal.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.tserver.log;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.accumulo.tserver.log.DfsLogger.LOG_FILE_HEADER_V4;
+import static org.apache.accumulo.tserver.logger.LogEvents.OPEN;
+
+import java.io.DataOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.accumulo.core.cli.Help;
+import org.apache.accumulo.core.crypto.CryptoUtils;
+import org.apache.accumulo.core.spi.crypto.NoFileEncrypter;
+import org.apache.accumulo.start.spi.KeywordExecutable;
+import org.apache.accumulo.tserver.logger.LogFileKey;
+import org.apache.accumulo.tserver.logger.LogFileValue;
+
+import com.beust.jcommander.IParameterValidator;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.google.auto.service.AutoService;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@AutoService(KeywordExecutable.class)
+public class CreateEmptyWal implements KeywordExecutable {
+  private static final LogFileValue EMPTY = new LogFileValue();
+
+  @SuppressFBWarnings(value = "PATH_TRAVERSAL_IN",
+      justification = "file output path provided by an admin")
+  // public to allow jcommander access
+  public static class DirValidator implements IParameterValidator {
+    @Override
+    public void validate(String name, String value) throws ParameterException {
+      Path path = Paths.get(value);
+      if (!Files.exists(path)) {
+        throw new ParameterException("Directory " + path.toAbsolutePath() + " does not exist");
+      }
+      if (!Files.isDirectory(path)) {
+        throw new ParameterException(path.toAbsolutePath() + " is not a directory");
+      }
+    }
+  }
+
+  static class Opts extends Help {
+    @Parameter(names = {"-d", "--dir"},
+        description = " <dir> the output directory, output will be [dir]/empty.wal",
+        required = true, validateWith = DirValidator.class)
+    String dirName;
+  }
```

Comment:
Rather than specify a directory and hard-code the file name, why not just allow the user to specify the file specifically?

Also, I'm not sure we need JCommander here at all... we should certainly fail if the specified output file exists already, but that's trivially done by passing `CREATE_NEW` option when the data stream is opened. We don't need all the complexity for validation, when the built-in behavior with simpler code already gets us the validation checks when we specify the output file.

Reply:
ab4e900992 - Changed to take file and delegate file existence check to use `CREATE_NEW` 

Kept JCommander.
  - Keeps utility similar to other keyword executable utilities.
  - Simplifies potential merging into CreateEmpty

Reply:
Keeping JCommander to work similarly to the RFile create empty command seems reasonable, but if the goal is to make it as similar as possible, it should not have a `-w` or `--wal-file` option, but instead accept a list of unnamed parameters without flags, as in: `<command> <path> { <path> ... }`.

Reply:
b07123b3af - removed named option in favor of unnamed parameter to proved file / path name,